src/tasks/deploy-sync.js

```diff
@@ -28,10 +29,16 @@ gulp.task('deploy:sync-init', () => {
   const environment = config.environment.split(/\s*,\s*|\s+/)[0];
 
   const envObj = tkConfig[environment];
+  const valid_id = utils.validateThemeId(envObj.theme_id);
   let proxyTarget = `https://${envObj.store}`;
 
-  if (envObj.theme_id && (envObj.theme_id === parseInt(envObj.theme_id, 10))) {
-    proxyTarget += `?preview_theme_id=${envObj.theme_id}`;
+  if (!valid_id) {
+    messages.invalidThemeId(envObj.theme_id, environment);
+    process.exit();
```

Comment:
instead of doing `process.exit()`, I would consider throwing an error here instead. `process.exit()` is dangerous because it just silently stops everything. In this case, we want to notify the user that something has gone wrong.